### DISCUSSION TITLE
How to programmatically disable check for updates?

### DISCUSSION THREAD

**Discussion by Rodrigodd** [visualboyadvance-m/visualboyadvance-m#1142] [Q&A] [answered]
Recently, I am trying to get Visual Boy Advance - M to work in [daid's GBEmulatorShootout](https://github.com/daid/GBEmulatorShootout). I got it working locally in [my fork](https://github.com/Rodrigodd/GBEmulatorShootout/tree/gameroy), but on GitHub CI, a modal window asking about automatically updates checking appear, blocking the emulator from running. [You can see the results in CI here](https://rodrigodd.github.io/GBEmulatorShootout/), where all tests for the VisualBoyAdvance-M fails.

I tried to look around at the code to see where the logic of the modal windows is, and therefore where the data that disables the window after it being addressed is stored, but I could not find it. It appears that in older versions that data was store in the config file, but not anymore.

So, is there any way to disable the modal window before the emulator is run the first time? (By, like, editing the config file, or changing some value in the registry)

**Comment by rkitover** [comment]
Pass `-DENABLE_ONLINEUPDATES=OFF` to cmake.

**Comment by ZachBacon** [comment]
There is no way to disable it other than at compile time via a cmake option, you can view current available compile options in the readme.md file

**Reply by Rodrigodd** [reply]
But how the emulator remembers to not show the modal window again after you click on it the first time?

**Reply by ZachBacon** [reply]
I already gave you the answer and you don't seem to understand it, essentially you have to recompile vba-m from source using the cmake options to disable online updates and then you won't get the dialog at all.

**Reply by Rodrigodd** [reply]
I did understand the answer, and I agree that that compiling option is a way to solve my problem. But currently, daid's test frameworks just downloads the emulator from GitHub releases, so building it from source would greatly complicate the process.

And the emulator already appears to have an internal logic for deciding if the modal windows should appear, because it stops appearing at startup after it is clicked for the first time, and that can only be done by storing that information somewhere. I want to know where that logic is, and if I can exploit it for disabling the modal window before starting the emulator for the first time.

**Reply by ZachBacon** [reply]
That's a limitation of winsparkle and how they handle things, we just use it.

**Reply by ZachBacon** [reply, marked as answer]
Looking at their wiki https://github.com/vslavik/winsparkle/wiki/Registry-Settings seems you can use this

**Reply by Rodrigodd** [reply]
Found it, thanks!
![image](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/51273772/2d30b830-e6ff-4d5e-8432-b9b9f88f15c1)
